dbt_semantic_interfaces/protocols/saved_query.py

```diff
@@ -0,0 +1,39 @@
+from abc import abstractmethod
+from typing import Optional, Protocol, Sequence
+
+from dbt_semantic_interfaces.protocols.metadata import Metadata
+from dbt_semantic_interfaces.protocols.where_filter import WhereFilter
+
+
+class SavedQuery(Protocol):
+    """Represents a query that the user wants to run repeatedly."""
+
+    @property
+    @abstractmethod
+    def metadata(self) -> Optional[Metadata]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def name(self) -> str:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def description(self) -> Optional[str]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def metric_names(self) -> Sequence[str]:  # noqa: D
+        pass
+
+    @property
+    @abstractmethod
+    def group_by_item_names(self) -> Sequence[str]:  # noqa: D
+        pass
```

Comment:
Separate question, why do we need this named in this way? Is this for internal developer ergonomics?

It's a little odd to me to take in group_bys as a list of strings and just return it from the more verbose protocol named field.

Also, how does this work with the consolidation we're trying to do around how people reference these things? Like what happens with `Dimension(.....)` calls? Do we want to make this conform to the interface we ultimately want instead of sticking with bare string references everywhere?

Reply:
Yeah, it was for developer ergonomics. The direction that @QMalcolm mentioned that we wanted to head to was decouple the protocol from the YAML. But seems like that's not the case. @QMalcolm mind elaborating?

Reply:
We do want to head the direction of decoupling the protocol from the YAML. Unfortunately `dbt-semantic-interfaces` does not currently have an architecture to support that. Specifically the json schema definition has to map to the pydantic implementation, and the pydantic implementation must satisfy the protocol. This architecture also leads us to marking things as optional in the protocol that are actually required, because the implementation has to mark them as optional (but we always ensure they're filled in with transformation rules).

Interestingly `dbt-core` does have an architecture that allows decoupling the yaml from the protocol. In dbt-core there is a concept of unparsed nodes and parsed nodes. YAML is loaded raw into the unparsed nodes. Then there exists mapping logic (similar to what DSI's transformation rules do) that create the "full" parsed node which satisfies the protocol. This is the direction I think we should move to in DSI, and I think it's out of scope for this PR.

The specific problem with this implementation is that it doesn't allow for setting `group_by_item_names` in deserialization. This actually ties us to the yaml because we'd actually have to output `group_bys` in the serialization, however that isn't part of the protocol so it shouldn't be part of the serialization. Could you dive into deeper what ergonomics are provided by having both `group_bys` and `group_by_item_names` as they both return the same thing currently?

Reply:
Got it. Regarding the naming - it was just for consistency with how things were named in MF. Given the situation, I'm fine removing `group_by_item_names`.